src/gausskernel/storage/access/k2/pg_gate_api.cpp

```diff
@@ -119,7 +128,25 @@ K2PgStatus PgGate_GetSharedCatalogVersion(uint64_t* catalog_version) {
 K2PgStatus PgGate_InitPrimaryCluster()
 {
   elog(DEBUG5, "PgGateAPI: PgGate_InitPrimaryCluster");
-  return K2PgStatus::NotSupported;
+
+  auto catalog = GetCatalog();
+  auto skvstat = catalog->InitPrimaryCluster();
+  if (skvstat.is2xxOK()) {
+      return K2PgStatus {
```

Comment:
use Status::OK or extend it to pass in the code

Reply:
Done.